packages/datastore/datastore/src/react/hooks/query.ts

```diff
@@ -9,6 +9,10 @@ const createSubscribeToMore = (model: Model, dispatch: Dispatch<Action>) => {
         // TODO subscribe to specific predicate
         const subscription = model.subscribe(eventType, (event) => {
             const newData = updateResult(event.data);
+            if (subscription.closed) {
+                // Important to check beacuse Componnent could be unmounted
```

Comment:
But this means that we have memory leaks and we hiding the error that is causing them?

Reply:
I don't think so. I added this check incase this listener is active when the subscription was ended. The listener won't be called for new events but what the listener that was just called before unsubscription